transactor/src/main/scala/io/mediachain/transactor/CborSerialization.scala

```diff
@@ -91,55 +92,77 @@ object CborSerialization {
       typeName <- getTypeName(cMap)
       deserializer <- Xor.fromOption(
         deserializers.get(typeName),
-        UnknownObjectType(typeName)
+        UnknownObjectType(typeName.toString)
       )
       value <- deserializer.fromCMap(cMap)
     } yield value
   }
 
 
-  /**
-    * String constants for the `type` field used to indicate the type of
-    * object represented by a cbor map.
-    */
-  object CBORTypeNames {
-    val Entity = "entity"
-    val Artefact = "artefact"
-    val EntityChainCell = "entityChainCell"
-    val ArtefactChainCell = "artefactChainCell"
-    val CanonicalEntry = "insert"
-    val ChainEntry = "update"
-    val JournalBlock = "journalBlock"
-
-
-    // TODO: Include subtypes in the sets below
-    /**
-      * The set of all valid ArtefactChainCell type names (including subtypes)
-      */
-    val ArtefactChainCellTypes: Set[String] = Set(
+  sealed trait CborTypeName {
+    val stringValue: String
+    def cborString: CString = CString(stringValue)
+    override def toString = stringValue
+  }
+
+  object CborTypeNames {
```

Comment:
@vyzo, it occurred to me this morning that MediachainType might be a better name...  CborType makes me think of e.g. CInt, CString, etc (the various cbor primitives).   What do you think?